src/Native/build-native.sh

```diff
@@ -64,8 +64,31 @@ check_native_prereqs()
 {
     echo "Checking pre-requisites..."
 
-    # Check presence of CMake on the path
-    hash cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script"; exit 1; }
+    # Check for CMake
+    # TODO: Remove the usage comment below
+    #   ./build.sh -StrictToolVersionMatch -- --OverrideScriptsFolderPath "/Users/raeda/Desktop/tools-local-copy/unix"
+    # Since run.exe cannot handle native-only build arguments, OverrideScriptsFolderPath is passed as additional string argument.
+    probeToolsArgs="--ToolName cmake"
+
+    if [ $__StrictToolVersionMatch -eq 1 ]; then
+        probeToolsArgs=$probeToolsArgs" -StrictToolVersionMatch"
+    fi
+
+    if [ ! -z "$__OverrideScriptsFolderPath" ]; then
+        probeToolsArgs="$probeToolsArgs --OverrideScriptsFolderPath $__OverrideScriptsFolderPath"
```

Comment:
I've seen this pattern a few places in other scripts/repos too, and it concerns me because it means the args can't have spaces in them. I think we need to decide if we actually want to support building under a path with spaces. And if it's just a future goal, if we're ok adding debt against it.

One way around it is to add literal quotes when building up the args variable and use `eval` to run the command. E.g. `args="$args \"$foo\""`, then, instead of `bar $args` or `bar "$args"` (neither of which work) do `eval "bar $args"` to completely reevaluate what's quoted. Another way would be saying we're taking a hard dependency on some bash stuff and use arrays.